.github/workflows/build-image.yml

```diff
@@ -8,13 +8,29 @@ on:
       - '*'
   pull_request:
 
+defaults:
+  run:
+    shell: 'bash -Eeuo pipefail -x {0}'
+
 jobs:
+  generate-matrix:
+    name: Generate Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.generate.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@master
+      - id: generate
+        name: Enumerate Dockerfiles
+        run: |
+          matrix="$(dirname */Dockerfile | sort -rn | jq -csR 'rtrimstr("\n") | split("\n") | { directory: . }')"
+          echo "::set-output name=matrix::$matrix"
   build-image:
+    needs: generate-matrix
     runs-on: ubuntu-latest
     strategy:
-      matrix:
-        variant: [ 'main', 'slim', 'main,threaded', 'slim,threaded' ]
-        perl-version: [ '5.032.001', '5.030.003' ]
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
```

Comment:
Maybe better to have a filter for currently supported versus old versions still (or just have the builds for older versions an new, separate workflow) - we really _don't_ have to rebuild everything all the time.

Reply:
>maybe we could have it only build directories whose contents have changed in the PR / commit that's being built?

Yep, we also try to check with this with the regeneration workflow (e.g. running a `diffstat` to see what changed.)

>Do you think it also makes sense to add a "supported" or "EOL" flag to the config so we can more clearly separate those?

Maybe we could also move out EOL versions to a different config entirely...

Reply:
Ooh, I like that -- what if we went a step further and made a totally separate `eol` directory to put them in, too?

So we'd have:

- `5.032.001-*/`
- ...
- `5.028.003-*/`
- ` config.yml`
- `eol/5.026.003-*/`
- ...
- `eol/config.yml`

Then the top-level directory listing would be much cleaner and more obvious what's supported vs best-effort?

Reply:
Yeah, that'd be awesome! :tada: